src/Essentials/src/Screenshot/Screenshot.ios.cs

```diff
@@ -31,66 +31,104 @@ public Task<IScreenshotResult> CaptureAsync(UIWindow window)
 		{
 			_ = window ?? throw new ArgumentNullException(nameof(window));
 
-			// NOTE: We rely on the window frame having been set to the correct size when this method is invoked.
-			UIGraphics.BeginImageContextWithOptions(window.Bounds.Size, false, window.Screen.Scale);
-			var ctx = UIGraphics.GetCurrentContext();
-
-			// ctx will be null if the width/height of the view is zero
-			if (ctx is not null && !TryRender(window, out _))
+			if (!OperatingSystem.IsIOSVersionAtLeast(17))
 			{
-				// TODO: test/handle this case
-			}
+				// NOTE: We rely on the window frame having been set to the correct size when this method is invoked.
+				UIGraphics.BeginImageContextWithOptions(window.Bounds.Size, false, window.Screen.Scale);
+				var ctx = UIGraphics.GetCurrentContext();
+
+				// ctx will be null if the width/height of the view is zero
+				if (ctx is not null && !TryRender(window, out _))
+				{
+					// TODO: test/handle this case
+				}
+
+				var image = UIGraphics.GetImageFromCurrentImageContext();
+				UIGraphics.EndImageContext();
 
-			var image = UIGraphics.GetImageFromCurrentImageContext();
-			UIGraphics.EndImageContext();
+				var result = new ScreenshotResult(image);
 
-			var result = new ScreenshotResult(image);
+				return Task.FromResult<IScreenshotResult>(result);
+			}
 
-			return Task.FromResult<IScreenshotResult>(result);
+			using var renderer = new UIGraphicsImageRenderer (window.Bounds.Size);
+			var resultImage = renderer.CreateImage ((UIGraphicsImageRendererContext imageContext) => {
+				if (!TryRender (window, out _))
+				{
+					// TODO: test/handle this case
+				}
+			});
+			var screenshotResult = new ScreenshotResult(resultImage);
+			return Task.FromResult<IScreenshotResult>(screenshotResult);
 		}
 
 		public Task<IScreenshotResult?> CaptureAsync(UIView view)
 		{
 			_ = view ?? throw new ArgumentNullException(nameof(view));
 
-			// NOTE: We rely on the view frame having been set to the correct size when this method is invoked.
-			UIGraphics.BeginImageContextWithOptions(view.Bounds.Size, false, view.Window.Screen.Scale);
-			var ctx = UIGraphics.GetCurrentContext();
-
-			// ctx will be null if the width/height of the view is zero
-			if (ctx is not null && !TryRender(view, out _))
+			if (!OperatingSystem.IsIOSVersionAtLeast(17))
```

Comment:
Can we use the new method here instead of the different mechanisms.

Reply:
@mattleibow, I have removed the old approach.